agent/src/com/cloud/agent/Agent.java

```diff
@@ -551,6 +555,10 @@ protected void processRequest(final Request request, final Link link) {
                         answer = setupAgentKeystore((SetupKeyStoreCommand) cmd);
                     } else if (cmd instanceof SetupCertificateCommand && ((SetupCertificateCommand) cmd).isHandleByAgent()) {
                         answer = setupAgentCertificate((SetupCertificateCommand) cmd);
+                    } else if (cmd instanceof CheckUrlCommand) {
```

Comment:
This will enable any agent (cpvm, ssvm etc) to handle the command, if we want only the KVM agent to handle these commands they should be moved to libvirt computing resource.